app/controllers/api/physical_servers_controller.rb

```diff
@@ -41,37 +41,77 @@ def restart_mgmt_controller_resource(type, id, _data)
     end
 
     def refresh_resource(type, id, _data = nil)
-      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
+      refresh_physical_server("refresh_ems", type, id)
+    end
 
-      ensure_resource_exists(type, id) if single_resource?
+    def apply_config_pattern_resource(type, id, data)
+      apply_config(:apply_config_pattern, type, id, data)
```

Comment:
Is this method going to be used in a similar way to `change_resource_state` where it will be reused with multiple states?